LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionCN470RP1.cs

```diff
@@ -66,20 +68,17 @@ public RegionCN470RP1()
         /// <param name="upstreamDataRate">The upstream data rate.</param>
         /// <param name="deviceJoinInfo">Join info for the device, if applicable.</param>
         /// </summary>
-        public override bool TryGetDownstreamChannelFrequency(double upstreamFrequency, out double downstreamFrequency, ushort? upstreamDataRate = null, DeviceJoinInfo deviceJoinInfo = null)
+        public override bool TryGetDownstreamChannelFrequency(Hertz upstreamFrequency, out Hertz downstreamFrequency, ushort? upstreamDataRate = null, DeviceJoinInfo deviceJoinInfo = null)
         {
             if (!IsValidUpstreamFrequency(upstreamFrequency))
                 throw new LoRaProcessingException($"Invalid upstream frequency {upstreamFrequency}", LoRaProcessingErrorCode.InvalidFrequency);
 
             var upstreamChannelNumber = (int)Math.Round(
-                (upstreamFrequency - StartingUpstreamFrequency) / FrequencyIncrement,
+                (upstreamFrequency - StartingUpstreamFrequency) / FrequencyIncrement.Units,
```

Comment:
I'm not sure I understand this expression. The dividend here is `Hertz` and divisor is a `double`. We don't explicitly define the / operator in `Hertz` so I'm not quite sure how this works; could you help me understand?

Reply:
Subtracting two frequencies returns the difference as a `long`:

```c#
public static long operator -(Hertz a, Hertz b) => checked((long)a.value - (long)b.value);
```

My thinking was that the difference between two frequencies isn't a frequency but just a number that represents the units of difference. So that expression is really a `long` being divided by a `double`. The whole idea of the primitives is that you can define the rules governing the correct use of its values without much ceremony.